woocommerce-services.php

```diff
@@ -357,14 +357,17 @@ public function load_textdomain() {
 		 * @codeCoverageIgnore
 		 */
 		public function init() {
+			$this->load_dependencies();
 			add_action( 'admin_init', array( $this, 'admin_enqueue_scripts' ) );
 
 			if ( ! WC_Connect_Options::get_option( 'tos_accepted', false ) ) {
 				add_action( 'admin_init', array( $this, 'admin_tos_notice' ) );
 				return;
+			} else if ( ! $this->check_jetpack_install() ) {
```

Comment:
Ouch, I missed that logic. You are totally right, I didn't erase the `wp_options` so I already had the ToS accepted.

I've changed it so the `Jetpack` notice appears first, and then simplified the `can_accept_tos()` logic since at that point Jetpack is already guaranteed to be installed.